Gems/EMotionFX/Code/Source/Editor/ObjectEditor.cpp

```diff
@@ -73,7 +73,18 @@ namespace EMotionFX
 
     void ObjectEditor::InvalidateAll()
     {
-        m_propertyEditor->InvalidateAll();
+        // If we Invalidate without giving the Search string, filtering Colliders will not work
+        // properly (nothing will be filtered out, instead only hightlighted)
+        // If we pass an empty filterstring, the Motion Id Picker will not be shown
+        // and AnimGraphUIFixture.CanAddMotionToAnimGraphNode will fail
```

Comment:
Thanks for the comments, that explains it quite clear.
That's said, I have two suggestions:
1. I don't think the ui editor code should know about the UI test exist at all. It's defeating the purpose for the test if the ui code need to be modified to make sure the test doesn't fail.
2. does this function fail if m_propertyEditor->InvalidateAll("")? (when you pass in an empty string)? in that case I'd rather patch it in the InvalidateAll function itself, rather than doing additional check here.

Reply:
In case InvalidateAll get's called with an empty string, the Motion Id Picker (but as far as i have seen nothing else) does not work.

Reply:
I'll remove the last line of the comment

Reply:
If InvalidateAll get called without parameter, it works tho? How is that different than calling it with empty string?